### PR TITLE
Joining invited member to empty room

### PR DESCRIPTION
**Pull request by kazakbai**
problem description here https://github.com/matrix-org/synapse/issues/3602

### CODE REVIEW

File: synapse/handlers/room_member.py [hunk 1]
```diff
@@ -888,7 +888,8 @@ def _is_host_in_room(self, current_state_ids):
             if not event:
                 continue
 
-            if event.membership == Membership.JOIN:
+            if (event.membership == Membership.JOIN
+                    or event.membership == Membership.INVITE):
```
Comment (richvdh) [inline synapse/handlers/room_member.py]: this doesn't look right. servers with no users in the `join` state won't receive updates in the room, so need to do the full `make_join` dance to get an up-to-date room DAG.

File: changelog.d/4232.bugfix [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Fix an issue where error {"errcode":"M_UNKNOWN","error":"No known servers"}
```
Comment (richvdh) [inline changelog.d/4232.bugfix]: this isn't a very meaningful description for the average user, I'm afraid.

File: synapse/handlers/room_member.py (hunk 1 above)
Comment (richvdh) [reply synapse/handlers/room_member.py]: I understand the problem. I'm saying this will break more than it fixes.

Comment (kazakbai) [reply synapse/handlers/room_member.py]: For example, what can break?

Comment (richvdh) [reply synapse/handlers/room_member.py]: you'll end up with out-of-date room state.

Comment (kazakbai) [reply synapse/handlers/room_member.py]: if 3 people are invited to the room, two accepted the invitation, and the third accepted later, after talking about something and leaving the room, the third should chat chat after accepting the invitation; give to the list of available

Comment (richvdh) [reply synapse/handlers/room_member.py]: I understand the problem. I'm saying this will break more than it fixes.

**Comment by richvdh** [comment]
as discussed, this doesn't seem the correct solution. Closing for now.

**Pull request closed without merging**